src/ducks/balance/History.jsx

```diff
@@ -1,32 +1,87 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 import cx from 'classnames'
+import * as d3 from 'd3'
 import { translate } from 'cozy-ui/react'
 import { Figure } from 'components/Figure'
+import { groupBy, sumBy, uniq } from 'lodash'
+import { format as formatDate, parse as parseDate } from 'date-fns'
+import LineChart from 'components/Chart/LineChart'
+import { getBalanceHistories } from './helpers'
 import styles from './History.styl'
 
 class History extends Component {
+  getCurrentBalance() {
+    return sumBy(this.props.accounts, a => a.balance)
+  }
+
+  getBalanceHistory() {
+    const { accounts, transactions } = this.props
+    const balanceHistories = getBalanceHistories(accounts, transactions)
+    const balanceHistory = Object.entries(balanceHistories.all)
+      .sort((a, b) => {
+        if (a[0] < b[0]) {
+          return 1
+        }
+
+        if (a[0] > b[0]) {
+          return -1
+        }
+
+        return 0
+      })
+      .map(([date, balance]) => ({
+        x: parseDate(date),
+        y: balance
+      }))
```

Comment:
All this work is done at each render, I think it should be done only when the data changes.